content/blog/arduino-esp-now-lib/index.md

```diff
@@ -0,0 +1,493 @@
+  Serial.println("  MAC Address: " + WiFi.macAddress());
+  Serial.printf("  Channel: %d\n", ESPNOW_WIFI_CHANNEL);
+
+  // Initialize the ESP-NOW protocol
+  if (!ESP_NOW.begin()) {
+    Serial.println("Failed to initialize ESP-NOW");
+    Serial.println("Reeboting in 5 seconds...");
+    delay(5000);
+    ESP.restart();
+  }
+
+  // Register the new peer callback that we created before
+  // The NULL parameter is an optional argument that can be passed to the callback.
+  ESP_NOW.onNewPeer(register_new_broadcaster, NULL);
+
+  Serial.println("Setup complete. Waiting for a broadcast message...");
+}
+
+void loop() {
+  delay(1000); // Delay to avoid watchdog reset
+}
+```
+
+After uploading the code to the receiver device, you can start the broadcaster device. The receiver device will automatically register the broadcaster device and start receiving the messages.
+
+For example:
+
+- Broadcaster Device:
+  {{< asciinema key=cast/broadcaster idleTimeLimit=2 speed=1 cols=80 rows=24 poster=npt:0:15 >}}
+
+- Receiver Device:
+  {{< asciinema key=cast/broadcast-receiver idleTimeLimit=2 speed=1 cols=80 rows=24 poster=npt:0:15 >}}
+
+As you can see, the receiver device automatically registers the broadcaster device and starts receiving the messages. If you add more broadcaster devices to the network, the receiver device will also register them and start receiving their messages. This way, you can easily create a network of devices using ESP-NOW.
+
+This example was based on the [`ESP_NOW_Broadcast_Master`](https://github.com/espressif/arduino-esp32/blob/master/libraries/ESP_NOW/examples/ESP_NOW_Broadcast_Master/ESP_NOW_Broadcast_Master.ino) and [`ESP_NOW_Broadcast_Slave`](https://github.com/espressif/arduino-esp32/blob/master/libraries/ESP_NOW/examples/ESP_NOW_Broadcast_Slave/ESP_NOW_Broadcast_Slave.ino) examples provided in the ESP-NOW library. You can find them in the Arduino IDE under `File > Examples > ESP_NOW`.
+
+## Advanced Examples
+
+To create more complex applications using ESP-NOW, you can explore the advanced [examples](https://github.com/espressif/arduino-esp32/blob/master/libraries/ESP_NOW/examples) provided in the ESP-NOW library. These examples demonstrate how to create a network of devices, send and receive data between multiple devices, handle different types of messages, add encryption to the communication, automatically decide the role of the device in the network without hardcoding it, and more.
+
+As the code for these examples is more extensive, we recommend checking the commented code directly in the Arduino IDE under `File > Examples > ESP_NOW`. You can find examples such as [`ESP_NOW_Network`](https://github.com/espressif/arduino-esp32/blob/master/libraries/ESP_NOW/examples/ESP_NOW_Network/ESP_NOW_Network.ino) that demonstrate how to create a network of devices with automatic role assignment and cryptography.
+
+The aim of this example is to demonstrate how to create a network of devices using the ESP-NOW protocol.
```

Comment:
Since this paragraph and the following ones until the end of the section talk about the `ESP_NOW_Network` example, suggest starting a new subsection here.

```suggestion
### `ESP_NOW_Network`

The aim of the `ESP_NOW_Network` example is to demonstrate how to create a network of devices using the ESP-NOW protocol.
```

Also, `demonstrate how to create a network of devices` is repeated from the previous paragraph. You might want to reduce duplication.